### PR TITLE
implementation of v1.18+ and v1.19+ for ingress

### PR DESCRIPTION
**Pull request by rowanruseler**
Should resolve: https://github.com/rowanruseler/helm-charts/issues/93

I did not yet test this

Signed-off-by: Rowan Ruseler <rowanruseler@gmail.com>

### CODE REVIEW

File: charts/pgadmin4/templates/ingress.yaml [hunk 1]
```diff
@@ -33,9 +36,18 @@ spec:
         paths:
         {{- range .paths }}
           - path: {{ .path }}
+          {{- if and .pathType (semverCompare ">=1.18-0" $kubeVersion) }}
+          pathType: {{ .pathType }}
+          {{- end }}
             backend:
+            {{- if semverCompare ">=1.19-0" $kubeVersion }}
```
Comment (qtomato3) [inline charts/pgadmin4/templates/ingress.yaml:43]: Missing `service`. Correct config looks like this
```
          - path: /
            pathType: Prefix
            backend:
              service:
                name: pg-pgadmin4
                port:
                  name: http
```
- either `.port.number` or `.port.name` is fine I guess.

**Comment by qtomato3** [comment]
Tested on k3s v1.20.6+k3s1 with above commented changes.

Could you add GitHub Actions to spin up a v1.19+ cluster to test against? 
From their [document](https://github.com/helm/kind-action), specific kind version can be used like this: 
- Your current [kind-action@v1.0.0](https://github.com/helm/kind-action/tree/v1.0.0 ) => kind version default to v0.8.1 => default node image is Kubernetes v1.18.2 (see [kind v0.8.0 release notes](https://github.com/kubernetes-sigs/kind/releases/tag/v0.8.0))
- `version`: The kind version to use (default: v0.9.0) -- helm/kind-action@v1.1.0

```
    steps:
      - name: Create k8s Kind Cluster
        uses: helm/kind-action@v1.1.0
        with:
          version: "v0.10.0"        # specify kind version
          cluster_name: "custom-name"
```


```
kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T11:56:40Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.6+k3s1", GitCommit:"8d0432824a9fd9474b67138b7630c33f285d332f", GitTreeState:"clean", BuildDate:"2021-04-16T19:04:44Z", GoVersion:"go1.15.10", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by rowanruseler** [comment]
@qtomato3 thank you for taking the time to review. If you have the time, could you double check it again? I adjusted all as requested, and thanks for pointing out I just missed a whole `service` part :)

**Comment by qtomato3** [comment]
Tested on k3s v1.20.6+k3s1 again and all is good.

**Comment by viceice** [comment]
Upgrade from 1.6.1 to 1.6.2 failed:
image

Those changes should be at least classified as minor update 😕

**Comment by viceice** [comment]
I've fixed it, but please do a minor version bump next time something will break existing users (better a major upgrade) so users are more aware that something important changed.

I'm not expecting a breaking change in a patch version bump.

**Comment by rowanruseler** [comment]
> I've fixed it, but please do a minor version bump next time something will break existing users (better a major upgrade) so users are more aware that something important changed.
> 
> I'm not expecting a breaking change in a patch version bump.

you are correct, my apologies.

**Pull request merged**